multiwallet_test.go

```diff
@@ -0,0 +1,35 @@
+package dcrlibwallet_test
+
+import (
+	"os"
+	"testing"
+	"testing/quick"
+
+	. "github.com/raedahgroup/dcrlibwallet"
+)
+
+// canUseDir returns true if the program can create the directory
+// if it doesn't exist or the directory is empty if it exists.
+// If will delete the directory if it is empty.
+func canUseDir(directory string) bool {
+	if os.MkdirAll(directory, os.ModePerm) == nil {
+		return os.Remove(directory) == nil
+	}
+	return false
+}
+
+// TestNewMultiWallet tests if NewMultiWallet will return an error for valid
+// paths or not return an error for invalid paths
+func TestNewMultiWallet(t *testing.T) {
+	f := func(rootDir string) bool {
+		canUse := canUseDir(rootDir)
+		_, err := NewMultiWallet(rootDir, "", "testnet")
+		if canUse {
+			return err == nil
```

Comment:
It is possible that the directory is usable (i.e. it exists or has been created if it doesn't exist) but some other error is returned. In such a case, this would return false and that would not be a representation that the directory is not usable.

Summary: do not expect error to be nil because the directory is usable, but expect err to NOT be nil if directory is not usable.

On another note, do we really even need this test? I believe test cases should be created from user stories and some defined acceptance criteria, not nit-picky.

Reply:
It's Quality Assurance, if it can be done, it will be done and it should behave in a expected manner.
What other error can there be?

Reply:
A simple scan of the `NewMultiwallet` function shows multiple sites where an error could be returned, theoretically speaking. Again, it is possible that the directory is usable (i.e. it exists or has been created if it doesn't exist) but some other error is returned.

Reply:
For the bit about QA, I respectfully disagree. The statement, "if it can be done, it will be done" does not and should not apply to writing tests. 
Doing so would easily amount to a waste of programmer time.

Imagine writing test to ensure that a function properly returns the sum of 2 numbers. Err, well maybe we should write such a test but like I said above, you don't test basically every little bit of code to ensure it works as expected. You write test cases following user stories and acceptance criteria.

Instead of writing a test to check `if NewMultiWallet will return an error for valid paths or not return an error for invalid paths`, we probably should be testing more relevant/sensitive methods and functions. Imo, a good test would be - `NewMultiWallet` should return an error for invalid arguments and no error for valid arguments. Testing specifically for path validity is nit-picky in my opinion.

That said, I have no reason to not merge this if the test code is corrected as requested in the initial review. I just think we can do better.

Reply:
Where are these user stories and acceptance criteria you speak of?

Reply:
Path validity is input validity. It should return an error for invalid arguments(cannot use the path argument) and no error for valid arguments (can use the path argument). That's exactly what the test checks...